File: VisitBosnia/VisitBosnia.Services/Database/Agency.cs
using System;
using System.Collections.Generic;

namespace VisitBosnia.Services.Database
{
    public partial class Agency
    {
        public Agency()
        {
            AgencyMembers = new HashSet<AgencyMember>();
            Events = new HashSet<Event>();
        }

        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string Phone { get; set; } = null!;
        public string? ResponsiblePerson { get; set; }
        public string Address { get; set; } = null!;
        public int CityId { get; set; }

        public virtual City City { get; set; } = null!;
        public virtual ICollection<AgencyMember> AgencyMembers { get; set; }
        public virtual ICollection<Event> Events { get; set; }
    }
}

File: VisitBosnia/visit_bosnia_mobile/lib/model/agency.dart
import 'city/city.dart';

class Agency {
  int? id;
  String? name;
  String? email;
  String? phone;
  String? responsiblePerson;
  String? address;
  int? cityId;
  City? city;

  Agency(
      {this.id,
      this.name,
      this.email,
      this.phone,
      this.responsiblePerson,
      this.address,
      this.cityId,
      this.city});

  Agency.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    email = json['email'];
    phone = json['phone'];
    responsiblePerson = json['responsiblePerson'];
    address = json['address'];
    cityId = json['cityId'];
    city = json['city'] != null ? new City.fromJson(json['city']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['email'] = this.email;
    data['phone'] = this.phone;
    data['responsiblePerson'] = this.responsiblePerson;
    data['address'] = this.address;
    data['cityId'] = this.cityId;
    if (this.city != null) {
      data['city'] = this.city!.toJson();
    }
    return data;
  }
}
